tensorpipe/common/defs.h

```diff
@@ -183,6 +183,12 @@ class LogEntry final {
   TP_THROW_ASSERT() << "Expected " << __TP_EXPAND_OPD(a) \
                     << " " TP_STRINGIFY(op) << " " << __TP_EXPAND_OPD(b)
 
+#define __TP_DCHECK_CUDA(a)                                            \
+  if (unlikely(cudaSuccess != (a)))                                    \
+  TP_THROW_ASSERT() << __TP_EXPAND_OPD(a) << " "                       \
+                    << cudaGetErrorName(cudaPeekAtLastError()) << " (" \
+                    << cudaGetErrorString(cudaPeekAtLastError()) << ")"
```

Comment:
Argument `a` is an expression, if we execute it again we will get a different error.

Reply:
Right, good point. If you see below, though, we should not put expressions in a DCHECK if we want them to be executed. This may allow us to use `a` and avoid `cudaPeekAtLastError`. Otherwise, we could have the macro assign `a` to a temporary variable and then use that...